module-2/lab-pivot-table-and-correlation/your_code/main.ipynb

```diff
@@ -21,13 +21,14 @@
   },
```

Comment:
- Bonus: "lo he intentado pero no he logrado nada en claro. He creado una columan con el maximo y pretendìa restarla a la del salario normal y hacer la media. También he intentado reemplzar con np.where los valores maximos por los minimos al haber sólo dos valores en cada departamento pero tampoco". **Pásame lo que hayas intentado y lo vemos el lunes, es más fácil de lo que parece. Al ser pocos registros despista.....**
- In this chunk of code, invert the axis, x minutes and x steps, it's a better picture.  
```Pythonfitbit.plot.scatter(x='Steps', y='Minutes Very Active')```
- Good one!! 
"What can you say about Minute Very Active and Steps? Write a comment below
Correlation function its around 0.57, what it looks like a strong positive correlation between both of them. As much as you walk the wrist band register these minutes as very active phisically. The scatter plot confirms it visually."
- In minutes sedentary vs steps the same as above, change the axis of variables. You say "If you just look to the scatter plot without regression line it seems like there is a negative realtionship between steps and minutes sedentary, so you can tend to imagine that if this person barely walks the registers for Minutes Sedenatry will be higher, and it fact that makes sense. But if you look further you can see that the coeficient is positive and nearly zero, so that implies that you cannot set a relation between both concepts. This is reinforced by the regression line which is almost plane" that is not quite correct. We cannot infer any relationship between minutes sedentary as we can't see any pattern in the data. This is confirmed by a corr coef near to 0. 
- What about the comments on this one? "Use matplotlib to visually plot MinutesOfSleep vs Steps. Based on the results of the computed correlation and the plot. What can you say about these 2 variables?"
- Better in 2 histograms than the 2 variables in the same one. It's easy to interpret. 
```Python 
#instead of 
time_grades.plot.hist()
#use 
time_grades.hist() 
```
- Good one!! 
"Use matplotlib to visually plot Study time vs Grade. Based on the results of the computed correlation and the plot. What can you say about these 2 variables?
Aparently, and even though the sample is reduced, it looks like there's a positive correlation between hpurs spent studying and the grade this students achieved. The most they studied, the highest grade they got." However, there is a point where it seems that higher number of study hours don't mean higher grades. This cut-off seems to be 8/9 study hours. 
- Regarding Spearman vs Pearson: 
Spearman vs Pearson: It is usually a little bit higher Spearman than Pearson but it doesn't mean necessarily that the relationship is not linear. Why? The Pearson is most appropriate for measurements taken from an interval scale (temperature, dates, lengths, etc), while the Spearman is best for measurements taken from ordinal scales (rank orders, spectrum of values (agree, neutral, disagree), or healthy vs non-healthy). However, with large samples, the correlation coefficients will be similar. The largest differences will be seen with small sample sizes. Pearson's assumes constant variance and linearity. If your variable violates that, Spearman's would be the best to use. This is the best explanation I have seen up to now (extract from quora). In other words, Spearman's coefficient measures the rank order of the points (it doesn't care where the points are) while Pearson measures the linear relationship between the two. In the end, it's not about a better or worse correlation, it's more about that we should use one or another depending on the characteristics of our variables.
In statistics and generally speaking, the fact two variables correlate cannot prove anything - only further research can actually prove that one thing affects the other.Data reliability is related to the size of the sample. The more data you collect, the more reliable your result.